iceoryx_binding_c/source/c_runtime.cpp

```diff
@@ -23,13 +23,31 @@ extern "C" {
 
 void iox_runtime_register(const char* const name)
 {
-    PoshRuntime::getInstance(name);
+    if (name == nullptr)
+    {
+        LogError() << "Application name is a nullptr!";
+        std::terminate();
+    }
+    else if (strnlen(name, iox::MAX_PROCESS_NAME_LENGTH + 1) > MAX_PROCESS_NAME_LENGTH)
```

Comment:
Is the check for length not already done in verifyInstanceName() when calling getInstance() in posh_runtime.cpp?